src/hooks/useQuota/useQuota.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState, useContext, useCallback, useEffect } from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { getQuota, NotEligibleError } from "../../services/quota";
+import { ProductContext } from "../../context/products";
+import { transform } from "lodash";
+import { Quota, CampaignPolicy } from "../../types";
+import { usePrevious } from "../usePrevious";
+
+type QuotaState = "DEFAULT" | "FETCHING_QUOTA" | "NO_QUOTA" | "NOT_ELIGIBLE";
+
+export type QuotaHook = {
+  quotaState: string;
+  quotaResponse: Quota | null;
+  allQuotaResponse: Quota | null;
+  updateQuota: () => void;
+  quotaError?: Error;
+};
+
+const filterQuotaWithAvailableProducts = (
+  quota: Quota,
+  products: CampaignPolicy[]
+): Quota => {
+  const filteredQuota: Quota = {
+    remainingQuota: []
+  };
+  transform(
+    quota.remainingQuota,
+    (result: Quota, itemQuota) => {
+      if (products.some(policy => policy.category === itemQuota.category))
+        result.remainingQuota.push(itemQuota);
+    },
+    filteredQuota
+  );
+
+  if (quota.globalQuota) {
+    filteredQuota.globalQuota = [];
+    transform(
+      quota.globalQuota!,
+      (result: Quota, itemQuota) => {
+        if (products.some(policy => policy.category === itemQuota.category))
+          result.globalQuota!.push(itemQuota);
+      },
+      filteredQuota
+    );
+  }
+
+  if (quota.localQuota) {
+    filteredQuota.localQuota = [];
+    transform(
+      quota.localQuota!,
+      (result: Quota, itemQuota) => {
+        if (products.some(policy => policy.category === itemQuota.category))
+          result.localQuota!.push(itemQuota);
+      },
+      filteredQuota
+    );
+  }
+
+  return filteredQuota;
+};
+
+const hasNoQuota = (quota: Quota): boolean => {
+  if (quota === null) {
+    return true;
+  }
+  return quota.remainingQuota.every(item => item.quantity === 0);
+};
+
+const hasInvalidQuota = (quota: Quota): boolean => {
+  // Note: Invalid quota refers to negative quota received
+  if (quota === null) {
+    return true;
+  }
+  return quota.remainingQuota.some(item => item.quantity < 0);
+};
+
+export const useQuota = (
+  ids: string[],
+  authKey: string,
+  endpoint: string
+): QuotaHook => {
+  const [quotaState, setQuotaState] = useState<QuotaState>("DEFAULT");
+  const [quotaResponse, setQuotaResponse] = useState<Quota | null>(null);
+  const [allQuotaResponse, setAllQuotaResponse] = useState<Quota | null>(null);
```

Comment:
What's the reason for using `null` rather than doing `useState<Quota>()`?

Reply:
I have changed this to:
```suggestion
  const [quotaResponse, setQuotaResponse] = useState<Quota>();
  const [allQuotaResponse, setAllQuotaResponse] = useState<Quota>();
```